articles/rules/metadata-in-rules.md

```diff
@@ -311,42 +338,131 @@ The resulting user is:
 }
 ```
 
-### Considerations
+## Client object metadata
 
-The metadata must be a valid JSON object and can not contain a dot in key field names in `user_metadata` or `app_metadata`.
+The sample rules below assume that the client is represented by the following JSON:
 
-The following is not allowed.
 ```js
 {
+  ...
+  "client_metadata": {
+  "key1": "valueforkey"
+  }
+...
+}
+```
+
+### Read
 
-"preference.color" : "pink"
+To read metadata from a rule, you only need to access the correct client property.
 
+```js
+function(client, context, callback){
+  client.client_metadata = client.client_metadata || {};
+  if (client.client_metadata.key1 === 'valueforkey'){
+  // this code would be executed
+  }
+...
 }
 ```
 
-The following is accepted, however.
+### Update
+
+To update a property, use the `push' method.
+
+```js
+function(client, context, callback){
+  client.client_metadata = client.client_metadata || {};
+  // update the client_metadata that will be part of the response
+  client.client_metadata.key1 = client.client_metadata.key1 || [];
+  client.client_metadata.key1.push('newValue');
+
+  // persist the client_metadata update
+  auth0.clients.updateClientMetadata(client.client_id, client.client_metadata)
+    .then(function(){
+      callback(null, client, context);
+    })
+    .catch(function(err){
+      callback(err);
+    });
+}
+```
+
+The resulting client is:
 
 ```js
 {
+...
+  "client_metadata": {
+  "key1": "newValue"
+  }
+...
+}
+```
+
+### Delete
 
-"color" : "light.blue"
+To delete a property, set the value to `null`.
```

Comment:
what about setting it to `undefined`? Will that have the same result? We should specify the behavior just to be sure.